backend/api/routes/users.js

```diff
@@ -46,18 +46,41 @@ router.route("/get-all").get(authenticateMiddleware, async (req, res) => {
       throw new Error("Error fetching the current user", userErr);
     }
 
-    //TODO: ONCE SAVED CONNECTIONS IS IMPLEMENTED (THE PAGE) we need to filter out and only display streamers the user has not connected with. Dont want to display current connections on the discover page.
-    //fetch all profile rows
-    const { data: streamers, streamersError } = await supabase
+    //get the connections this user has. We get connections from connections table and select the connected streamer id whenever that connection's user_id matches ours
+    const { data: connections } = await supabase
+      .from("connections")
+      .select("connected_streamer_id")
+      .eq("user_id", userId);
```

Comment:
What happen if the connection is on the opposite direction (i.e. userId is on connected_streamer_id)?